### ISSUE TITLE
Returns receiver checker issues spurious errors on method references that should return `@This`

### ISSUE THREAD

**Issue by kelloggm** [kelloggm/object-construction-checker#79]
The error looks something like this:

```
/Users/kelloggm/Research/lombok-testing/java-webauthn-server/webauthn-server-core/build/delombok/main/com/yubico/webauthn/RelyingParty.java:291: error: [methodref.receiver.bound.invalid] Incompatible receiver type
        startAssertionOptions.getUserVerification().ifPresent(pkcro::userVerification);
                                                              ^
  Method
    @MaybeThis PublicKeyCredentialRequestOptionsBuilder
  is not a valid method reference for
    @BottomThis PublicKeyCredentialRequestOptionsBuilder userVerification(@BottomThis PublicKeyCredentialRequestOptionsBuilder this, @MaybeThis UserVerificationRequirement p0) in com.yubico.webauthn.data.PublicKeyCredentialRequestOptions.PublicKeyCredentialRequestOptionsBuilder
  found   : @MaybeThis PublicKeyCredentialRequestOptionsBuilder
  required: @BottomThis PublicKeyCredentialRequestOptionsBuilder
```

You can see the error in the `rr-methodref` branch.

**Comment by msridhar** [comment]
I believe I ran into this before and @smillst suggested the problem is https://github.com/typetools/checker-framework/issues/948.  Good to have a small test case to reproduce.  I'm not sure there is a fix we can do in our checkers.  @smillst any thoughts?

**Comment by smillst** [comment]
According to that email thread, `framework/tests/all-systems/Issue450.java` is a simple test case.  I looked into this and I think the problem is different than https://github.com/typetools/checker-framework/issues/948.  I think it has something to do with polymorphic quals on receiver of member references.   I started working on fixing this is https://github.com/smillst/checker-framework/tree/polyRef.  The test all pass in that branch, but I didn't add a test case, so I'm not sure if that fixed this problem.

**Comment by msridhar** [comment]
I will try to see if that Checker Framework branch fixes the issue later tonight.

**Comment by msridhar** [comment]
@smillst I confirmed that with your `polyRef` branch, the test on the `rr-methodref` branch in this repo passes!

**Comment by smillst** [comment]
I created a test case using the Tainting Checker and made a pull request against typetools:
https://github.com/typetools/checker-framework/pull/2695
